pool/connection_pool.go

```diff
@@ -170,7 +172,7 @@ func ConnectWithOpts(addrs []string, connOpts tarantool.Opts, opts Opts) (*Conne
 	connPool.state.set(connectedState)
 
 	for _, s := range connPool.addrs {
-		go connPool.controller(s)
+		go connPool.controller(ctx, s)
```

Comment:
But we call `tryConnect` (calls `Connect`) and `reconnect` from `controller`. Do we need to pass `context.Background` to them?

Reply:
> Do we need to pass `context.Background` to them?

`controller(ctx)` -> `tryConnect(ctx)` should be something more interesting. It would be great to cancel `controller(ctx)` -> `tryConnect(ctx)` with `ConnectionPool.Close()`.